active-rfcs/0033-router-navigation-failures.md

```diff
@@ -148,27 +149,49 @@ router.currentRoute // { path: '/login', redirectedFrom: { path: '/profile/dashb
 Since `router.afterEach` also triggers when a navigation fails, we need a way to know if the navigation succeeded or failed. To do that, we introduce an extra parameter that contains the same _failure_ we could find in a resolved navigation:
 
 ```js
-import { NavigationFailureType } from 'vue-router'
+import { NavigationFailureType, isNavigationFailure } from 'vue-router'
 
 router.afterEach((to, from, failure) => {
-  if (failure) {
-    if (failure.type === NavigationFailureType.canceled) {
-      // ...
-    }
+  if (isNavigationFailure(failure)) {
+    // ...
   }
 })
 ```
 
-# Drawbacks
+## Differentiating Navigation failures
 
-- Breaking change although migration is relatively simple and in many cases will allow the developer to remove existing code
+Instead of directly checking the `type` property, we can use the `isNavigationFailure` helper:
+
+```js
+import { NavigationFailureType, isNavigationFailure } from 'vue-router'
+
+router.afterEach((to, from, failure) => {
+  // Any kind of navigation failure
+  if (isNavigationFailure(failure)) {
+    // ...
+  }
+  // Only duplicated navigations
+  if (isNavigationFailure(failure, NavigationFailureType.duplicated)) {
+    // ...
+  }
+  // Aborted or canceled navigations
+  if (
+    isNavigationFailure(
+      failure,
+      NavigationFailureType.aborted | NavigationFailureType.canceled
```

Comment:
Nit pick but this should be `NavigationFailureType.aborted || NavigationFailureType.canceled` (double pipe)?

Reply:
No, it's a bitwise OR. People can still do

```js
isNavigationFailure(failure, NavigationFailureType.aborted) || isNavigationFailure(failure, NavigationFailureType.aborted)
```

I think we could show this one first and say in a note we can use the bitwise operator as well. I use the bitwise one inside the codebase to make it smaller

Reply:
Ahhh I see thanks! Good to know 😳

Reply:
I unresolved the question in case other people want to provide their input too 🙂